### ISSUE TITLE
[mlx_lm] Use logits_processors in speculative_generate_step

### ISSUE THREAD

**Issue by mattjcly** [ml-explore/mlx-examples#1255]
Right now, `logits_processors` is unused in `speculative_generate_step` https://github.com/ml-explore/mlx-examples/blob/9c2ef38d4d7a6c6afc1f9c90cb26cadb95f7343e/llms/mlx_lm/utils.py#L335 so things like repetition penalty and whatever else one may want to do in a logits processor is not being applied when using speculative decoding.
